lib/fs.js

```diff
@@ -531,8 +531,22 @@ function read(fd, buffer, offset, length, position, callback) {
 ObjectDefineProperty(read, internalUtil.customPromisifyArgs,
                      { value: ['bytesRead', 'buffer'], enumerable: false });
 
+// usage:
+// fs.readSync(fd, buffer, offset, length, position);
+// OR
+// fs.readSync(fd, buffer, {}) or fs.readSync(fd, buffer)
 function readSync(fd, buffer, offset, length, position) {
   validateInt32(fd, 'fd', 0);
+
+  if (arguments.length <= 3) {
+    // Assume fs.read(fd, buffer, options)
+    const options = offset || {};
+
+    offset = options.offset || 0;
+    length = options.length || buffer.length;
+    position = options.position;
+  }
+
```

Comment:
```suggestion
  if (arguments.length <= 3)
    // Assume fs.read(fd, buffer, options)
    ({ offset = 0, length = buffer.length, position } = options || {})
```

Reply:
BTW, I think it shouldn't use the default value when user passing null on options

```js
foo = options.foo || 1
// output 1 when `options.foo` is null

({ foo } = options)
// still null when passing
```

Reply:
#32479

Reply:
updated